src/Oscoin/P2P/Types.hs

```diff
@@ -32,12 +50,50 @@ import qualified Network.Gossip.IO.Peer as Gossip (Peer)
 import           Codec.Serialise (Serialise)
 import qualified Codec.Serialise as CBOR
 import           Data.Aeson (FromJSON, ToJSON)
+import           Data.Char (isAlphaNum)
 import           Data.Hashable (Hashable(..))
 import           Data.IP (IP(..))
+import           Data.Profunctor (Profunctor(dimap))
+import           Data.String (IsString(..))
+import qualified Data.Text as T
+import           Data.Vector (Vector)
+import qualified Data.Vector as V
 import           Formatting as F
 import qualified Generics.SOP as SOP
-import           Network.Socket (HostName, PortNumber, SockAddr)
+import qualified Network.DNS as DNS
+import           Network.Socket (PortNumber, SockAddr)
+import qualified Network.Socket as Network
+
+-- | A logical oscoin network.
+data Network =
```

Comment:
This type should probably live somewhere else, as it is conceivable that other things may depend on it's value (in combination with `Environment`, supposedly)

Reply:
Brainstorming names here -- something like `Oscoin.Node.Infrastructure` ?

Reply:
https://github.com/oscoin/oscoin/issues/427#issuecomment-478059758